src/services/uniswapV3/LiquidityService.ts

```diff
@@ -0,0 +1,415 @@
+    const ticksQuery = JSON.stringify({
+      query: `{ ticks(
+            where: {poolAddress: "${poolAddress.toLowerCase()}", liquidityNet_not: "0"}
+            first: 1000,
+            orderBy: tickIdx,
+            orderDirection: asc,
+            skip: ${skip}
+          ) {
+            tickIdx
+            liquidityGross
+            liquidityNet
+          }
+        }
+      `,
+    });
+
+    const response = await axios.post('https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3', ticksQuery, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    return response.data.data.ticks;
+  }
+
+  getFeedParams(data: LeavesAndFeeds) {
+    const {fcdsFeeds, leavesFeeds} = data;
+
+    if (Object.keys(fcdsFeeds).length === 0 && Object.keys(leavesFeeds).length === 0) {
+      throw new Error('[LiquidityService] No leaves to process');
+    }
+
+    const feedsToFetch = [];
+
+    for (const [i, feed] of Object.entries(leavesFeeds)) {
+      const uniswapV3Feed = feed.inputs.find((entry) => {
+        return entry.fetcher.name === 'UniswapV3';
+      });
+
+      if (uniswapV3Feed) {
+        const {chainFrom, token0, token1} = uniswapV3Feed.fetcher.params as UniswapV3Param;
+        feedsToFetch.push({chainFrom, token0, token1});
+      }
+    }
+
+    if (feedsToFetch.length === 0) {
+      throw new Error('[LiquidityService] No feeds to fetch');
+    }
+
+    return feedsToFetch;
+  }
+
+  async fetchLiquidityTicks(
+    poolAddress: string,
+    ticks: GraphTick[] = [],
+    limit = 1000,
+    skip = 0,
+  ): Promise<Promise<GraphTick[]> | GraphTick[]> {
+    console.log('START');
+    const response = await axios.post<{data: LiquidityTick}>(
+      'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
+      {
+        query: `{ ticks(
+              where: {poolAddress: "${poolAddress.toLowerCase()}", liquidityNet_not: "0"}
+              first: ${limit},
+              skip: ${skip},
+              orderBy: tickIdx,
+              orderDirection: asc
+            ) {
+              tickIdx
+              liquidityGross
+              liquidityNet
+            }
+          }`,
+      },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+
+    if (response.data.data.ticks.length === limit) {
+      const concatTicks = ticks.concat(response.data.data.ticks);
+      return await this.fetchLiquidityTicks(poolAddress, concatTicks, limit, limit + skip);
+    }
+
+    return ticks.concat(response.data.data.ticks);
+  }
+
+  async getPoolTickSum(poolAddress: string, token0: Token, token1: Token, fee: FeeAmount, chainId: ChainsIds) {
+    const provider = this.blockchainProviderRepository.get(chainId);
+    const poolContract = new ethers.Contract(poolAddress, IUniswapV3PoolABI.abi, provider);
+    const [slot0, liquidity, graphTicks] = await Promise.all([
+      poolContract.slot0(),
+      poolContract.liquidity(),
```

Comment:
TODO for me:
- add to helper methods for slot0 and liquidity, so we can fetch it in one call. 
- or provide info how to use multicall contract

Reply:
this is done, but let's apply in next PR